### PR TITLE
Add support for optional OpenCage request params

### PR DESCRIPTION
**Pull request by KevinFarrell**

### CODE REVIEW

**Comment by nchaulet** [comment]
I am going to look at this, this week end

**Comment by nchaulet** [comment]
Could you add a little documentation in readme ?

File: lib/geocoder/opencagegeocoder.js [hunk 1]
```diff
@@ -28,7 +28,29 @@ OpenCageGeocoder.prototype._geocode = function (value, callback) {
...
+  if (value.q) {
+    if (value.bounds) {
+      if (Array.isArray(value.bounds)) {
+        params.bounds = value.bounds.join(',');
+      }
+      else {
+        params.bounds = value.bounds;
+      }
+    }
+    if (value.countrycode) {
```
Comment (nchaulet) [inline lib/geocoder/opencagegeocoder.js]: Could you rename it `countryCode`

File: lib/geocoder/opencagegeocoder.js [hunk 2]
```diff
@@ -28,7 +28,29 @@ OpenCageGeocoder.prototype._geocode = function (value, callback) {
...
+      params.countrycode = value.countrycode;
+    }
+    if (value.limit) {
+      params.limit = value.limit;
+    }
+    if (value.min_confidence) {
+      params.min_confidence = value.min_confidence;
+    }
+    params.q = value.q;
```
Comment (nchaulet) [inline lib/geocoder/opencagegeocoder.js]: GoogleGeocoder allow the same comportement but the q parameter is named `address` could you rename it to be coherent

File: lib/geocoder/opencagegeocoder.js [hunk 3]
```diff
@@ -28,7 +28,29 @@ OpenCageGeocoder.prototype._geocode = function (value, callback) {
...
+        params.bounds = value.bounds;
+      }
+    }
+    if (value.countrycode) {
+      params.countrycode = value.countrycode;
+    }
+    if (value.limit) {
+      params.limit = value.limit;
+    }
+    if (value.min_confidence) {
```
Comment (nchaulet) [inline lib/geocoder/opencagegeocoder.js]: I choose to use camelCase in node-geocoder could you rename it `minConfidence`

**Comment by KevinFarrell** [comment]
Parameter names updated and added a small snippet to the README.

**Comment by nchaulet** [comment]
Thanks :+1:

**Comment by r3wt** [comment]
awesome!

**Pull request merged**